docs/semgrep-ci/manual-ci-setup.md

```diff
@@ -0,0 +1,415 @@
+
+#### Jenkins CI code snippet
+
+This code snippet uses Jenkins declarative syntax.
+
+```javascript
+pipeline {
+  agent any
+    stages {
+      stage('Semgrep-Scan') {
+	      environment { 
+          SEMGREP_RULES = 'p/default'
+        } 
+        steps {
+          sh 'pip3 install semgrep'
+          sh 'semgrep ci'
+      }
+    }
+  }
+}
+```
+
+### BitBucket Pipelines
+
+To add Semgrep into your GitLab CI/CD pipeline, do the following steps.
+
+1. Create or edit a `bitbucket-pipelines.yml` configuration file to add
+   a Semgrep `step` as part of your pipeline. Refer to the [BitBucket Pipelines code snippet](#bitbucket-pipelines-code-snippet).
+2. Optional: You can also review the [Bitbucket guidelines guide](https://support.atlassian.com/bitbucket-cloud/docs/configure-bitbucket-pipelinesyml/) to help with editing the configuration file.
+3. Commit the configuration file into the root folder within the target repository. The Semgrep job starts automatically upon detecting the `bitbucket-pipelines.yml` commit in the previous step.
+4. Optional: Re-run or view the Pipeline by clicking from **your repository > Pipelines**.
+5. Customize the job's behavior, such as its rules to scan and files to ignore.
+
+:::note
+These steps can also be performed through BitBucket's UI wizard. This UI wizard can be accessed through **BitBucket > your repository > Pipelines > Create your first pipeline**.
+:::
+
+#### BitBucket Pipelines code snippet
+
+```yaml
+image: atlassian/default-image:latest
+
+pipelines:
+  default:
+    - parallel:
+      - step:
+          name: 'Run Semgrep scan with current branch'
+          deployment: dev
+          image: returntocorp/semgrep
+          script:
+            - export SEMGREP_RULES="p/default" # TODO get confirmation from Holden
+            - semgrep ci # TODO confirm that this will work
+```
+
+### CircleCI
+
+To add Semgrep into your CircleCI pipeline, do the following steps.
+
+1. Create or edit a `config.yml` configuration file to add a Semgrep command as
+   part of your CircleCI workflow. Refer to the [CircleCI code snippet](#circleci-code-snippet).
+2. Optional: You can also review the [CirclCI YAML introduction](https://circleci.com/docs/introduction-to-yaml-configurations) to help with editing the configuration file.
+3. Commit the configuration file into the `/.circleci` folder within the target
+   repository.
+4. The Semgrep job starts automatically upon detecting the `config.yml` commit in the previous step.
+5. Customize the job's behavior, such as its rules to scan and files to ignore.
+
+#### CircleCI code snippet
+
+```yaml
+version: 2.1
+jobs:
+  semgrep-scan:
+    parameters:
+      repo_path:
+        type: string
+		    # Change This to your repository path.
+			  # Sara: Isn't there a way to fetch these values from Circle CI so it's not hardcoded?
+			  # TODO Check with Holden.
+        default: myorg/semgrep-test-repo
+      default_branch:
```

Comment:
@holdenmcg Our current docs have this `myorg/semgrep-test-repo` - shouldn't the CI provider be feeding this such as << parameters.default_soundthing >>?

Reply:
Update. I checked all the parameters in the the the [CircleCI docs](https://circleci.com/docs/variables#pipeline-values) and it does not look like there's an easy way to do this (seems like I can't sed to extract the path/reponame either). @holdenmcg guess this means it's a dead end and we have to make the user do it manually?

Reply:
Ah, I have an update for this; there is a line we can use so they don't have to set this manually. I will update the docs with the new config; we don't need them to set any kind of repo_path variable.


In this case however, you can just remove lines 283 - 286 (the entire repo_path parameter), since we're using p/default and the repo path variable is for connecting to the app. 

So I will separately update the sample CI config/app setup page with the correct info, but for now you can remove that bit.

Reply:
Done. Thank you Holden.